src/OrleansRuntime/Placement/DeploymentLoadPublisher.cs

```diff
@@ -178,6 +178,11 @@ private void NotifyAllStatisticsChangeEventsSubscribers(SiloAddress silo, SiloRu
 
         public void SiloStatusChangeNotification(SiloAddress updatedSilo, SiloStatus status)
         {
+            this.ScheduleTask(() => { Utils.SafeExecute(() => this.OnSiloStatusChange(updatedSilo, status), this.logger); }).Ignore();
```

Comment:
I do think that calling via some proxy is a better solution. The only problem is that `GrainReference` can only be used from an addressable context today (another `Grain` or `SystemTarget`) when running within the silo.

Maybe that restriction can be relaxed and we can convert all of the code to make proxied calls. The problem is that it's not relaxed today and I feel like that's a larger change than we could stomach in a patch release. That kind of thing requires more thought - I'm sure the exception which caused this issue was not added lightheartedly, it's likely there to make sure scheduling contexts are being used correctly.